model/gen-ai/registry.yaml

```diff
@@ -177,6 +177,71 @@ groups:
               brief: 'Output tokens (completion, response, etc.)'
         brief: The type of token being counted.
         examples: ['input', 'output']
+      - id: gen_ai.agent.id   # alternatives: assistant (openai)
+        stability: experimental
+        type: string
+        brief: The unique identifier of the GenAI agent.
+        examples: ['asst_5j66UpCpwteGg4YSxUnt7lPY']
+      - id: gen_ai.agent.name
+        stability: experimental
+        type: string
+        brief: Human-readable name of the GenAI agent provided by the application.
+        examples: ["Math Tutor", "Fiction Writer"]
+      - id: gen_ai.agent.description
+        stability: experimental
+        type: string
+        brief: Free-form description of the GenAI agent provided by the application.
+        examples: ["Helps with math problems", "Generates fiction stories"]
+      - id: gen_ai.agent.tools
+        stability: experimental
+        type: string[]
+        brief: List of tool names available to the agent.
+        examples:
+          - ["Flights", "Search", "Code"]
+          - ["calc"]
+      - id: gen_ai.agent.status
+        stability: experimental
+        type:
+          members:
+            - id: completed
+              value: "completed"
+              brief: 'The agent has been completed'
+              stability: experimental
+            - id: failed
+              value: "failed"
+              brief: 'The agent was unable to complete the task due to an error.'
+              stability: experimental
+        brief: The status of the agent.
+      - id: gen_ai.agent.tool.name
+        stability: experimental
+        type: string
+        brief: Name of the tool utilized by the agent.
+        examples: ["Flights"]
+      - id: gen_ai.agent.tool.call.id
+        stability: experimental
+        type: string
+        brief: The tool call identifier.
+        examples: ['call_mszuSIzqtI65i1wAUOE8w5H4']
+      - id: gen_ai.agent.tool.type
+        stability: experimental
+        type: string
+        brief: Type of the tool utilized by the agent.
```

Comment:
can we add some details? I'm struggling to understand what is it (e.g. `extension` or `datastore` tool), where does this classification come from?

How would it map to tool types used by some popular agents?

Reply:
This is from https://www.kaggle.com/whitepaper-agents , I think it is a good summary. 

Like for langchain, here is a list of tools https://python.langchain.com/docs/integrations/tools/ , extensions focus for some integrations, like SerpAPI、Wikipedia API etc, function focus for some customized function, while datastore can help you for some RAG applications, hope this helps.

Reply:
I will add a note here as below:

- Extension: A tool executed on the agent-side to directly call external APIs, bridging the gap between the agent and real-world systems.
- Function: A tool executed on the client-side, where the agent generates parameters for a predefined function, and the client executes the logic.
- Datastore: A tool used by the agent to access and query structured or unstructured external data for retrieval-augmented tasks or knowledge updates.